flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -827,6 +829,8 @@ else if (transitionState(current, ExecutionState.FAILED, t)) {
                     }
                     // else fall through the loop and
                 }
+
+                cleanUpRegistry.close();
```

Comment:
NOTE! there is a change in behaviour. Now any exception from the cleanup will not be suppressed any more but will be treated as fatal error (I think correctly).